Missing preprocessor directives in Canvas.cc for when libjpeg isn't available

I broke this in #1152 probably.

```
../src/Canvas.cc: In static member function ‘static void Canvas::ToJpegBufferAsync(uv_work_t*)’:
../src/Canvas.cc:209:90: error: ‘write_to_jpeg_buffer’ was not declared in this scope
   write_to_jpeg_buffer(closure->canvas->surface(), closure, &closure->data, &closure->len);
```

--- a/src/Canvas.cc
+++ b/src/Canvas.cc
@@ -202,12 +202,14 @@ Canvas::ToPngBufferAsync(uv_work_t *req) {
     closure);
 }
 
+#ifdef HAVE_JPEG
 void
 Canvas::ToJpegBufferAsync(uv_work_t *req) {
   JpegClosure* closure = static_cast<JpegClosure*>(req->data);
 
   write_to_jpeg_buffer(closure->canvas->surface(), closure, &closure->data, &closure->len);
 }
+#endif
 
 /*
  * EIO after toBuffer callback.

